### PR TITLE
Modify chain consensus parameters to be height aware

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

**Comment by rnicoll** [comment]
Opening early so you can see progress, but this is no further along than it happens to compile.

File: doc/build-unix.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ Optional dependencies:
  Library     | Purpose          | Description
  ------------|------------------|----------------------
  miniupnpc   | UPnP Support     | Firewall-jumping support
- libdb5.1    | Berkeley DB      | Wallet storage (only needed when wallet enabled)
+ libdb4.8    | Berkeley DB      | Wallet storage (only needed when wallet enabled)
```
Comment (langerhans) [inline doc/build-unix.md]: Not blocking, but those need to be reverted if I'm not mistaken.

File: src/chainparams.cpp [hunk 2]
```diff
@@ -178,19 +207,32 @@ static CMainParams mainParams;
...
+        consensus.fPowAllowDigishieldMinDifficultyBlocks = false;
         consensus.nSubsidyHalvingInterval = 100000;
-        // XXX: BIP heights and hashes all need to be updated to Dogecoin values
-        consensus.BIP34Height = 21111;
-        consensus.BIP34Hash = uint256S("0x0000000023b3a96d3484e5abb3755c413e7d41500f8e2a5c3f0dd01299cd8ef8");
-        consensus.BIP65Height = 581885; // 00000000007f6655f22f98e72ed80d8b06dc761d5da09df0fa1dc4be4f861eb6
-        consensus.BIP66Height = 330776; // 000000002104c8c45e99a8853285a3b592602a3ccde2b832481da85e9e4ba182
+        consensus.nMajorityEnforceBlockUpgrade = 1500;
+        consensus.nMajorityRejectBlockOutdated = 1900;
+        consensus.nMajorityWindow = 2000;
```
Comment (langerhans) [inline src/chainparams.cpp]: Same comment here, on master those are 501, 750, 1000

**Comment by rnicoll** [comment]
It's based on #1403, but they're split to make them a bit easier to review.

**Comment by rnicoll** [comment]
This is running through a further full resync after rebase, but I'm fairly happy with just merging this now if it's okay?

**Pull request merged**